packages/flutter_tools/lib/src/web/devfs_web.dart

```diff
@@ -347,17 +341,20 @@ class WebDevFS implements DevFS {
         'web',
         'dart_stack_trace_mapper.js',
       ));
+      final String entrypoint = PackageUriMapper(mainPath, '.packages', null, null)
```

Comment:
Is this always a path under `lib`? What does `PackageUriMapper` do for files not under lib?

Reply:
If it's not under lib, it returns null - so I'm falling back to mainPath. That ... kinda works, what does the frontend_server produce in this case?

Reply:
Frontend_server will fall back to either the multi-root or absolute file uri (multi-root takes precedence).

The problem is down below we do `/packages/$entrypoint.lib.js` which won't work with the absolute or multi root uris.

Reply:
Ahh yeah, I'd need to check for the whole thing being null. I'll update with some tests